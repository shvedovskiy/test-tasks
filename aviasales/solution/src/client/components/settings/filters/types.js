// @flow
export type StopsType = {
  [stop: string]: boolean,
};
