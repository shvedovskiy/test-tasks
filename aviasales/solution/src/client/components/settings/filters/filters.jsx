// @flow
import * as React from 'react';
import { connect } from 'react-redux';

import { getAllStops } from 'src/store/rootSelectors';
import { changeStopsFilter } from 'src/store/settings/actions';
import type { State } from 'src/store/types';
import Stops from './stops/stops';
import type { StopsType } from './types';


type Props = {|
  render: (string, React.Element<typeof Stops>) => React.Node,
  stops: StopsType,
  changeStops: Function,
|}

export const Filters = ({ render, stops, changeStops }: Props) => {
  const stopsTitle = 'Количество пересадок';
  const child = <Stops stops={stops} changeStops={changeStops} />;

  return render(stopsTitle, child);
};

const mapStateToProps = (state: State) => ({
  stops: getAllStops(state),
});

export default connect(mapStateToProps, { changeStops: changeStopsFilter })(Filters);
