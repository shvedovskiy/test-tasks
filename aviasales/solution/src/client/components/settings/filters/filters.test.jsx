/* eslint-disable no-undef */
import React from 'react';

import { Filters } from './filters';


describe('<Filters />', () => {
  const props = {
    render: jest.fn(),
    stops: {
      '0': true,
      '1': false,
      '2': false,
    },
    changeStops: jest.fn(),
  };
  const filters = shallow(<Filters {...props} />);

  it('renders properly', () => {
    expect(filters).toMatchSnapshot();
  });
});
