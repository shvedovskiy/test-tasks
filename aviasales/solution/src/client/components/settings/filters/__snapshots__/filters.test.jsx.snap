// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filters
    changeStops={[MockFunction]}
    render={
      [MockFunction] {
        "calls": Array [
          Array [
            "Количество пересадок",
            <Stops
              changeStops={[MockFunction]}
              stops={
                Object {
                  "0": true,
                  "1": false,
                  "2": false,
                }
              }
            />,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    stops={
      Object {
        "0": true,
        "1": false,
        "2": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
